Sources/_StringProcessing/Engine/MEBuiltins.swift

```diff
@@ -194,9 +190,7 @@ extension String {
     isStrictASCII: Bool,
     isScalarSemantics: Bool
   ) -> String.Index? {
-    guard currentPosition < endIndex else {
-      return nil
-    }
+    assert(currentPosition < endIndex)
```

Comment:
Out of curiosity, why hoist these out? We'll probably want a naming convention for the assumption in this case, such as `atAssumingInBounds currentPosition`.

Reply:
As written, the check can't be done in this method, since we're missing the processor's `end`. With the `limitedBy` parameter we can add the guard back in.